### PR TITLE
JDK 16 support

### LINKED ISSUE #566: Build fails on JDK 16
The build works when using JDKs 1.8, 11, & 15, however there are test errors and the build fails when using JDK 16.

It looks like there are two different types of errors:
```
[INFO] Running org.yarnandtail.andhow.junit5.ext.KillAndHowBeforeAllTestsExtTest
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.007 s <<< FAILURE! - in org.yarnandtail.andhow.junit5.ext.KillAndHowBeforeAllTestsExtTest
[ERROR] org.yarnandtail.andhow.junit5.ext.KillAndHowBeforeAllTestsExtTest.completeWorkflow  Time elapsed: 0.004 s  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class org.mockito.internal.creation.cglib.ClassImposterizer$3
        at org.yarnandtail.andhow.junit5.ext.KillAndHowBeforeAllTestsExtTest.setUp(KillAndHowBeforeAllTestsExtTest.java:54)
```
and
```
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.17 s <<< FAILURE! - in org.yarnandtail.andhow.junit5.ext.KillAndHowBeforeEachTestExtensionTest
[ERROR] org.yarnandtail.andhow.junit5.ext.KillAndHowBeforeEachTestExtensionTest.completeWorkflow  Time elapsed: 0.139 s  <<< ERROR!
java.lang.ExceptionInInitializerError
        at org.yarnandtail.andhow.junit5.ext.KillAndHowBeforeEachTestExtensionTest.setUp(KillAndHowBeforeEachTestExtensionTest.java:50)
Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make protected final java.lang.Class java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain) throws java.lang.ClassFormatError accessible: module java.base does not "opens java.lang" to unnamed module @3f0aaff5
        at org.yarnandtail.andhow.junit5.ext.KillAndHowBeforeEachTestExtensionTest.setUp(KillAndHowBeforeEachTestExtensionTest.java:50)
```

Is there a simple fix for this or does it require a larger effort??

To compile with JDK16, an 'enforcer' rule in the root pom file needs to be changed:
```
<plugin>
  <groupId>org.apache.maven.plugins</groupId>
  <artifactId>maven-enforcer-plugin</artifactId>
  ...
  <version>[1.8.0,17)</version> (was 16)
  ...
</plugin>
```
This rule is there because it was known that JDK16 wasn't working, but I didn't know why.  it would be nice to figure that out so the build can happen on the latest JDK and this rule can be opened as above.

My Environment:
OS:  Mac 11.3.1 w/ M1 Processor
Java: openjdk version "16.0.1" 2021-04-20 & openjdk version "16.0.2" 2021-07-20
Maven: 3.8.1 & 3.8.2

### PR DESCRIPTION
**Pull request by alex-kar**
Fixes #566

* Replace `mockito-all` with latest `mockito-core`

### All Submissions:

Have you checked that...
* [x] Your pull request is to the *_main_* branch
* [x] Your code is up to date with the latest code from the *_main_*
* [x] You followed the guidelines in our [Developer Guide](https://sites.google.com/view/andhow/developer)?
* [x] Your code does not decrease test coverage (maybe it improves coverage!!)
* [x] If this is related to an issue, please include a link to the issue with the 'Fixes #XXX' syntax.

### CODE REVIEW

File: andhow-junit5-extensions/pom.xml [hunk 1]
```diff
@@ -75,7 +75,8 @@
 		</dependency>
 		<dependency>
 			<groupId>org.mockito</groupId>
-			<artifactId>mockito-all</artifactId>
+			<artifactId>mockito-core</artifactId>
+			<version>3.12.4</version>
```
Comment (eeverman) [inline andhow-junit5-extensions/pom.xml:79]: This dependency can just be deleted.  Its inherited from the parent pom, which you already fixed.

**Review by eeverman — CHANGES_REQUESTED** [review]
One minor change to remove my unneeded dependency, but this is a great fix.  I was worried that something had changed in the JVM that required changes in the project - glad that is not the case.
Nice fix!

File: andhow-junit5-extensions/pom.xml (hunk 1 above)
Comment (alex-kar) [reply andhow-junit5-extensions/pom.xml]: Fixed

**Review by eeverman — APPROVED** [review]

**Comment by eeverman** [comment]
Looks great.  I prefer to let people merge their own code - Please merge if you are happy with it.

**Pull request merged**